docs/src/getting-started/configuration.md

```diff
@@ -0,0 +1,183 @@
+| `--endpoint` | Faucet endpoint (IP:PORT) | - | Yes |
+| `--node-url` | Miden node RPC endpoint | - | Yes |
+| `--account` | Path to faucet account file | - | Yes |
+| `--network` | Network configuration | `localhost` | No |
+| `--timeout` | RPC request timeout | `5s` | No |
+| `--max-claimable-amount` | Max claimable base units per request | `1000000000` | No |
+| `--store` | SQLite store path | `faucet_client_store.sqlite3` | No |
+
+### Proof of Work Configuration
+
+| Option | Description | Default | Required |
+|--------|-------------|---------|----------|
+| `--pow-secret` | Secret for PoW challenges | - | No |
+| `--pow-baseline` | Base PoW difficulty (0-32) | `12` | No |
+| `--pow-challenge-lifetime` | Challenge validity duration | `30s` | No |
+| `--pow-cleanup-interval` | Cache cleanup interval | `2s` | No |
+| `--pow-growth-rate` | Difficulty growth rate | `1` | No |
+
+### Advanced Configuration
+
+| Option | Description | Default | Required |
+|--------|-------------|---------|----------|
+| `--remote-tx-prover-url` | Remote transaction prover | - | No |
+| `--api-keys` | Comma-separated API keys | - | No |
+| `--enable-otel` | Enable OpenTelemetry | `false` | No |
+
+## Environment Variables
+
+All configuration options can be set using environment variables:
+
+```bash
+# Basic configuration
+export MIDEN_FAUCET_ENDPOINT=http://localhost:8080
+export MIDEN_FAUCET_NODE_URL=https://rpc.testnet.miden.io:443
+export MIDEN_FAUCET_ACCOUNT_PATH=./faucet.mac
+export MIDEN_FAUCET_NETWORK=testnet
+
+# Proof of Work
+export MIDEN_FAUCET_POW_SECRET=your-secret-here
+export MIDEN_FAUCET_POW_BASELINE=12
+export MIDEN_FAUCET_POW_CHALLENGE_LIFETIME=30s
+export MIDEN_FAUCET_POW_GROWTH_RATE=1
+
+# Advanced
+export MIDEN_FAUCET_MAX_CLAIMABLE_AMOUNT=1000
+export MIDEN_FAUCET_TIMEOUT=10s
+export MIDEN_FAUCET_ENABLE_OTEL=true
+export MIDEN_FAUCET_API_KEYS=key1,key2,key3
+```
+
+## Network Configurations
+
+### Predefined Networks
+
+#### Localhost
+```bash
+--network localhost
+```
+- **Explorer URL**: Not available
+- **Address Display**: mlcl
+- **Use Case**: Local development
+
+#### Devnet
+```bash
+--network devnet
+```
+- **Explorer URL**: Not available
+- **Address Display**: mdev
+- **Use Case**: Development testing
+
+#### Testnet
+```bash
+--network testnet
+```
+- **Explorer URL**: https://testnet.midenscan.com/
+- **Address Display**: mtst
+- **Use Case**: Integration testing
+
+### Custom Network
+```bash
+--network custom
+```
+
+- **Explorer URL**: Not available
+- **Address Display**: mcst
+- **Use Case**: Run your custom network
+
+## Proof of Work Configuration
+
+### Understanding PoW
+
+The Proof of Work system prevents abuse by requiring computational work:
+
+1. **Baseline Difficulty**: Starting difficulty when no requests are pending
+2. **Growth Rate**: How quickly difficulty increases with load
+3. **Challenge Lifetime**: How long challenges remain valid
+4. **Cleanup Interval**: How often expired challenges are removed
```

Comment:
I would probably move these explanations up to the "Proof of Work" configuration section.